### PR TITLE
Physics state uses quantity

### PR DESCRIPTION
**Pull request by elynnwu**
## Purpose

#211 

## Code changes:

- state.py: remove physics specific restart 
- physics_state.py: all storages are now quantity, added init from storages for translate test
- microphysics.py: microphysics state also uses quantity
- translate_microphysics.py: make physics state from storages

## Checklist
Before submitting this PR, please make sure:

- [x] You have followed the coding standards guidelines established at [Code Review Checklist](https://drive.google.com/file/d/1R0nqOxfYnzaSdoYdt8yjx5J482ETI2Ft/view?usp=sharing).
- [ ] Docstrings and type hints are added to new and updated routines, as appropriate
- [ ] All relevant documentation has been updated or added (e.g. README, CONTRIBUTING docs)
- [ ] For each public change and fix in `pace-util`, HISTORY has been updated
- [ ] Unit tests are added or updated for non-stencil code changes

Additionally, if this PR contains code authored by new contributors:

- [ ] The names of all the new contributors have been added to CONTRIBUTORS.md

### CODE REVIEW

File: physics/pace/physics/physics_state.py [hunk 1]
```diff
@@ -212,36 +319,55 @@ def __post_init__(
...
         )
 
+    @classmethod
+    def init_from_storages(
+        cls,
+        storages: Mapping[str, Any],
+        sizer: pace.util.GridSizer,
+        quantity_factory: pace.util.QuantityFactory,
+        active_packages: List[str],
+    ):
```
Comment (jdahm) [inline physics/pace/physics/physics_state.py:339]:
```suggestion
    ) -> "PhysicsState":
```

File: physics/pace/physics/physics_state.py [hunk 2]
```diff
@@ -212,36 +319,55 @@ def __post_init__(
...
+        inputs = {}
```
Comment (jdahm) [inline physics/pace/physics/physics_state.py:340]:
```suggestion
        inputs: Dict[str, pace.util.Quantity] = {}
```
Will also need to import `Dict` from typing at the top of the module.

**Review by jdahm — APPROVED** [review]
Great! It's a nice step forward to use Quantity consistently in Pace.

**Comment by elynnwu** [comment]
launch jenkins

**Comment by elynnwu** [comment]
launch jenkins

**Pull request merged**